Scripts/Actuators/Thruster.cs

```diff
@@ -0,0 +1,107 @@
+// Copyright 2022 Laboratory for Underwater Systems and Technologies (LABUST)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using Marus.Logger;
+using Marus.Utils;
+using UnityEditor;
+using UnityEngine;
+namespace Marus.Actuators
+{
+
+    public class Thruster : MonoBehaviour
+    {
+        //Calback for change in voltage
+        public float lastForceRequest;
+        public float timeSinceForceRequest = 0.0f;
+
+        Rigidbody _vehicleBody;
+        Transform _vehicle;
+        GameObjectLogger<PwmLogRecord> _logger;
+        public ThrusterAsset[] thrusters;
+        public ThrusterAsset selectedThruster;
+        public AnimationCurve currentCurve = new AnimationCurve();
+        public int previousThrusterIndex = -1;
+        public int selectedThrusterIndex = 0;
```

Comment:
This may not be here. Only thing used here is the selectedThruster asset.

This class should be standalone and not depend on Editor. This is because Editor class will not be built when we build the project and this class will.

What needs to be done here is Asset loading similar to the GetAllInstances<> method on the edior. I suggest to create a method `GetAssetInstance<ThrusterAsset>(string name)` to load dynamically the asset in the Start() mehod of this class. The name here should be given from the editor, so create an additional string field of this class which will be set by the editor.